src/safeposix/syscalls/fs_calls.rs

```diff
@@ -1409,68 +1481,101 @@ impl Cage {
                             }
                         }
 
+                        // For `CharDev` type inode, the reading happens from the Character Device
+                        // file, with each device type returning different results returned
+                        // from the `_read_chr_file` file.
                         Inode::CharDev(char_inode_obj) => {
+                            // reads from character devices by matching the device number (DevNo) of
+                            // the DeviceInode.
                             self._read_chr_file(&char_inode_obj, buf, count)
                         }
 
+                        // For `Socket` type inode, a panic is returned as socket type files are not
+                        // supported.
                         Inode::Socket(_) => {
                             panic!("read(): Socket inode found on a filedesc fd.")
```

Comment:
A line within the panic function saying that read is not supported on sockets would be good here

Reply:
fixed.